addons/hr_attendance/models/hr_attendance_overtime.py

```diff
@@ -18,7 +18,7 @@ def _default_employee(self):
         required=True, ondelete='cascade', index=True)
     company_id = fields.Many2one(related='employee_id.company_id')
 
-    date = fields.Date(string='Day')
+    date = fields.Date(string='Day', required=True, index=True)
```

Comment:
I'm not sure that's possible? the upgrade script would have to give an arbitrary date since afaik we have no to tell what it should be so maybe we shouldn't make the field required?